# This parameter list is used to configure sensors for the Studica Robotics ROS2 control server.
# 
# To enable a sensor type, set "enabled: true" under its section.
# If the sensor type (e.g., cobra, duty_cycle) has a "sensors" list, you can use multiple sensors by uniquely naming them.
# Then, for each named sensor, add a section underneath with its specific parameters (e.g., port, topic).
#
# Example (ultrasonic):
#   sensors: ["ultrasonic1", "ultrasonic2"]
#   ultrasonic1:
#     ping: 8
#     echo: 9
#     topic: "range_1"
#   ultrasonic2:
#     ping: 10
#     echo: 11
#     topic: "range_2"
#

control_server:
  ros__parameters:
    cobra:
      enabled: false
      sensors: ["cobra"]
      cobra:
        vref: 5.0
        topic: "cobra"
      
    duty_cycle:
      enabled: false
      sensors: ["duty_cycle"]
      duty_cycle:
        port: 8
        topic: "duty_cycle_encoder"

    diff_drive_component:
      enabled: true
      name: "diff_drive_controller"
      can_id: 42
      motor_freq: 15600
      ticks_per_rotation: 1470
      left_port: 2
      right_port: 3
      invert_left: false
      invert_right: false
      wheel_radius: 0.05 # in metres
      wheel_separation: 0.28 # in metres
    
    diff_drive_odometry:
      name: "differential_odom"
      use_imu: false
      imu_topic: "imu"
      topic: "odom"

    dio:
      enabled: false
      sensors: ["dio"]
      dio:
        pin: 15
        type: "output" # "input" or "output"
        topic: "dio_state"
    
    encoder:
      enabled: false
      sensors: ["encoder"]
      encoder:
        port_a: 8
        port_b: 9
        topic: "encoder"

    imu:
      enabled: true
      name: "imu"
      topic: "imu"

    mecanum_drive_component:
      enabled: false
      name: "mecanum_drive_controller"
      can_id: 42
      motor_freq: 15600
      ticks_per_rotation: 1470
      wheel_radius: 0.05 # in metres
      wheelbase: 0.28 # in metres
      width: 0.3 # in metres
      front_left: 2
      front_right: 1
      rear_left: 0
      rear_right: 3
      invert_front_left: false
      invert_front_right: false
      invert_rear_left: false
      invert_rear_right: false

    mecanum_drive_odometry:
      name: "mecanum_odom"
      use_imu: true
      imu_topic: "imu"
      topic: "odom"

    servo:
      enabled: false
      sensors: ["servo"]
      servo:
        port: 14
        type: "standard" # "standard", "continuous", or "linear"
        topic: "servo_angle"

    sharp:
      enabled: false
      sensors: ["sharp"]
      sharp:
        port: 22
        topic: "ir_range"

    titan:
      enabled: true
      sensors: ["titan"]
      titan:
        can_id: 42
        motor_freq: 15600
        topic: "titan_encoders"

    ultrasonic:
      enabled: false
      sensors: ["ultrasonic"]
      ultrasonic:
        ping: 10
        echo: 11
        topic: "ultrasonic_range"
